R/options.R

```diff
@@ -1,270 +1,130 @@
-#' @return char vec
-#' @export
-#' @inherit .DollarNames.DataFrame return
-#' @keywords internal
-.DollarNames.polars_option_list = function(x, pattern = "") {
-  with(x, ls(pattern = pattern))
-}
-#' @export
-`print.polars_option_list` = function(x, ...) {
-  print("pl$options (polars_options, list-like with value validation):")
-  print(pl$get_polars_options())
-}
-
-#' #' @export
-#' `as.list.polars_option_list` = function(x, ...) {
-#'   pl$get_polars_options()
-#' }
-#' #' @export
-#' `as.vector.polars_option_list` = function(x, ...) {
-#'   as.vector(pl$get_polars_options())
-#' }
-
-
-
-
-#' @param ... any options to modify
-#'
-#' @param return_replaced_options return previous state of modified options
-#' Convenient for temporarily swapping of options during testing. The immediate
-#' return value is invisible.
+#' @return
+#' `pl$options` returns a named list with the value (`TRUE` or `FALSE`) of
+#' each option.
+#' `pl$set_options()` silently modifies the options values.
+#' `pl$reset_options()` silently resets the options to their default values.
 #'
-#' @rdname polars_options
-#' @name set_polars_options
-#' @return current settings as list
-#' @details setting an options may be rejected if not passing opt_requirements
 #' @examples
-#' pl$set_polars_options(strictly_immutable = FALSE)
-#' pl$get_polars_options()
-#'
+#' pl$set_options(maintain_order = TRUE, strictly_immutable = FALSE)
+#' pl$options
 #'
-#' # setting strictly_immutable = 42 will be rejected as
+#' # these options only accept booleans (TRUE or FALSE)
 #' tryCatch(
-#'   pl$set_polars_options(strictly_immutable = 42),
+#'   pl$set_options(strictly_immutable = 42),
 #'   error = function(e) print(e)
 #' )
 #'
-pl$set_polars_options = function(
-    ...,
-    return_replaced_options = TRUE) {
-  # check opts
-  opts = list2(...)
-  if (is.null(names(opts)) || !all(nzchar(names(opts)))) stopf("all options passed must be named")
-  unknown_opts = setdiff(names(opts), names(polars_optenv))
-  if (length(unknown_opts)) {
-    stopf(paste("unknown option(s) was passed:", paste(unknown_opts, collapse = ", ")))
-  }
-
-  # update options
-  replaced_opts_list = list()
-  for (i in names(opts)) {
-    opt_requirements = polars_optreq[[i]]
-    stopifnot(
-      !is.null(opt_requirements),
-      is.list(opt_requirements),
-      all(sapply(opt_requirements, is.function)),
-      all(nzchar(names(opt_requirements)))
-    )
-
-    for (j in names(opt_requirements)) {
-      opt_check = opt_requirements[[j]]
-      opt_value = opts[[i]]
-      opt_result = opt_check(opt_value)
-      if (!isTRUE(opt_result)) {
-        stopf(paste(
-          "option:", i, " must satisfy requirement named", j,
-          "with function\n", paste(capture.output(print(opt_check)), collapse = "\n")
-        ))
-      }
+#' # reset options to their default value
+#' pl$reset_options()
+
+pl$set_options = function(
+    strictly_immutable = TRUE,
```

Comment:
I think this default-value, but not really a default-value, would also surprise R users.
 How about `arg = NULL` to alias with not modified? 
 or not have any defaults and use `missingArg` to alias with not modfied?

Reply:
> too unexpected for R users.

I don't follow, it's classic to have default values for arguments so I don't understand why it would be unexpected. Also these default values don't appear in the autocompletion (cf screenshot above) but they appear in the docs, which I think is important.

Reply:
ok, I accept :)